src/main/java/net/fabricmc/loader/impl/launch/FabricMixinBootstrap.java

```diff
@@ -79,7 +76,104 @@ public static void init(EnvType side, FabricLoaderImpl loader) {
 		}
 
 		MixinBootstrap.init();
-		getMixinConfigs(loader, side).forEach(FabricMixinBootstrap::addConfiguration);
+
+		Map<String, ModContainerImpl> configToModMap = new HashMap<>();
+
+		for (ModContainerImpl mod : loader.getModsInternal()) {
+			for (String config : mod.getMetadata().getMixinConfigs(side)) {
+				ModContainerImpl prev = configToModMap.putIfAbsent(config, mod);
+				if (prev != null) throw new RuntimeException(String.format("Non-unique mixin config name %s used by %s and %s", config, prev.getMetadata().getId(), mod.getMetadata().getId()));
+
+				Mixins.addConfiguration(config);
+			}
+		}
+
+		for (Config config : Mixins.getConfigs()) {
+			ModContainerImpl mod = configToModMap.get(config.getName());
+			if (mod == null) continue;
```

Comment:
Whats the reason for this loop?

Reply:
I don't patch config addition in Mixin anymore, instead I add the config files normally, then pull the config objects back out, reassociate them with mods and add fabric specific data to them.